src/cpu.c

```diff
@@ -102,3 +138,27 @@ inline cycles load16(cpu* c, cpu_register_t reg, u16 value) {
 
   return 2;
 }
+
+inline cycles trans16(cpu* c, cpu_register16_t reg1, cpu_register16_t reg2) {
+  if (reg1 == reg2)
+    return 0;
+
+  u16 v1 = fetchReg(c, reg1);
+  u16 v2 = fetchReg(c, reg2);
+  setReg(c, reg1, v2);
+  setReg(c, reg2, v1);
+
+  return 2;
+}
+
+inline cycles trans8(cpu* c, cpu_register8_t reg1, cpu_register8_t reg2) {
+  if (reg1 == reg2)
+    return 0;
+
+  u8 v1 = fetchReg(c, reg1);
+  u8 v2 = fetchReg(c, reg2);
+  setReg(c, reg1, v2);
+  setReg(c, reg2, v1);
+
+  return 2;
+}
```

Comment:
TRANS is unfortunately not "exchange values in the register", it is "get from one register and put it into the other".

You are also going to need to set/reset the negative and zero flags for the changed register, as applicable.